packages/db-kit/src/cli/decodeTransaction/Inputs.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState, useEffect } from "react";
+import { Box, Text } from "ink";
+import { UncontrolledTextInput } from "ink-text-input";
+import Spinner from "ink-spinner";
+import type { Transaction, TransactionReceipt } from "web3-core";
+import type TruffleConfig from "@truffle/config";
+
+import { useTransactionInfo } from "./useTransactionInfo";
+
+export interface Props {
+  config: TruffleConfig;
+  onSubmit: (inputProps: {
+    transactionHash: string;
+    transaction: Transaction;
+    receipt: TransactionReceipt;
+    addresses: string[];
+  }) => void;
+}
+
+export const DecodeTransactionInputs = ({ config, onSubmit }: Props) => {
+  const [transactionHash, setTransactionHash] = useState<string | undefined>();
+  const { transaction, receipt, addresses } = useTransactionInfo({
+    config,
+    transactionHash
+  });
+
+  useEffect(() => {
+    if (transactionHash && transaction && receipt && addresses) {
+      onSubmit({
+        transactionHash,
+        transaction,
+        receipt,
+        addresses
+      });
+    }
+  }, [transaction, addresses]);
+
+  const spinners = [
```

Comment:
I found this section a bit difficult to follow. Looks like you have 0 up to a max of 2 elements in this array?

Reply:
That's right. Probably worth making a `<Spinners />` component or somesuch, eh?

Reply:
Can this be a later problem?

Reply:
Yea works for me